index.template.html

```diff
@@ -3916,6 +3925,191 @@ <h5>Inferencing of Personally Identifiable Information Privacy Risk</h5>
+             <a>Thing Model</a> definitions MUST use the JSON member 
+            name <code>required</code> at interaction level 
+            (<code>properties</code>, <code>actions</code>, and <code>events</code>)
+            to provide the name of the interaction in a string-based array list.
+        </span>    
+    </p>
+    
+    <p>    
+    The following sample shows the usage of <code>required</code> for the <a>Property</a> interaction <code>status</code>
+    and <a>Action</a> interaction <code>toggle</code>. 
+    </p>
+    
+    <pre class="example" title="MyLampThingModel serialized in JSON">
+    {
+        "@context": ["https://www.w3.org/2019/wot/td/v1"], 
+        "@type" : "ThingModel",
+        "title": "Lamp Thing Model",
+        "description" : "Lamp Thing Description Model",
+        "properties": {
+            "status": {
+                "description" : "current status of the lamp (on|off)",
+                "type": "string",
+                "readOnly": true
+            },
+            "required": ["status"]
+        },
+        "actions": {
+            "toggle": {
+                "description" : "Turn the lamp on or off"
+            },
+            "required": ["toggle"]
+        },
+        "events": {
+            "overheating": {
+                "description" : "Lamp reaches a critical temperature (overheating)",
+                "data": {"type": "string"}
+            }
+        }
+    }
+    </pre>
+    
+    <p>
+    Since the <a>Event</a> <code>overheating</code> is not manditory it may be not available  in a <a>Thing Description</a> 
+    instance.
+    </p>
+
+    <p>
+     The same Thing Model can be implemented by <a>Things</a> from multiple vendors.
+    A <a>Thing</a> can implement multiple Thing Model, define additional metadata
+    (vendor, location, security) and define bindings to concrete protocols.
+    To avoid conflicts between properties, actions and events from different Thing Model
+    that are combined into a common <a>Thing</a>, all these identifiers must be unique within a <a>Thing</a>.
+    </p>
+    
+    <p>
+    A common Thing Model for a class of devices enables writing applications across vendors and
+    creates a more attractive market for application developers.
+    A concrete Thing Description can implement multiple Thing Model and
+    thus can aggregate function blocks into a combined device.
+    </p><p>
+    The business models of cloud vendors are typically built on managing thousands of identical devices.
+    All devices with the same Thing Description Template can be managed in the same way by cloud services and applications.
+    It is easy to create multiple simulated devices, if the interface and the instance are treated separately.
+    </p>
+    <p>
+        Since a Thing Description Template is a subset of the Thing Description in which some optional and mandatory <a>Vocabulary Terms</a> 
```

Comment:
Thing Description Template -> Thing Model.   Would be a good idea to do a search for "Template" to see if you've caught them all.

Reply:
ok

Reply:
done